src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -454,12 +148,12 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
         const auto debugGroup(parameters.renderPass->createDebugGroup("opaque"));
 
         uint32_t i = 0;
-        for (auto it = renderItems.rbegin(); it != renderItems.rend(); ++it, ++i) {
+        for (auto it = layerRenderItems.rbegin(); it != layerRenderItems.rend(); ++it, ++i) {
```

Comment:
This sounds like a good use case for a `std::for_each` with a functor that would keep the state for `i`.

```
struct OpaquePass {
  void operator()(RenderItem&) { ...}
  int currentLayer{0};
};
```

Reply:
wouldn't it be an overkill (given that we've several passes |=> would need several functors) ? :thinking: